msal/src/main/java/com/microsoft/identity/client/PublicClientApplicationConfiguration.java

```diff
@@ -481,8 +481,8 @@ private void nullConfigurationCheck(String configKey, String configValue) {
 
     private boolean isBrokerRedirectUri() {
         final String BROKER_REDIRECT_URI_REGEX = "msauth://" + mAppContext.getPackageName() + "/.*";
-        final Pattern pairRegex = Pattern.compile(BROKER_REDIRECT_URI_REGEX);
-        final Matcher matcher = pairRegex.matcher(mRedirectUri);
+        final Pattern pairRegex = Pattern.compile(BROKER_REDIRECT_URI_REGEX.toLowerCase());
+        final Matcher matcher = pairRegex.matcher(mRedirectUri.toLowerCase());
```

Comment:
Mm.  I'm kinda suspicious here, but if you want to do this you can make your pattern case insensitive with Pattern.compile(BROKER_REDIRECT_URI_REGEX, Pattern.CASE_INSENSITIVE) and don't do all the string conversion.  Can we store this compiled pattern somewhere?

Reply:
Let me try that option. Should work too.

Reply:
And we should surely add a test for this behavior. If this is going to be something like 'com.example.application', should we escape the . characters, or are we ok with com.exampletapplication matching?